packages/Angular/ask-skip/src/lib/skip-chat/skip-chat.component.ts

```diff
@@ -453,21 +453,21 @@ export class SkipChatComponent implements OnInit, AfterViewInit, AfterViewChecke
     dc.UserID = md.CurrentUser.ID;
     if (await dc.Save()) {
       // now create a data context item for the linked record if we have one
-      if (this.LinkedEntityID && this.LinkedEntityID > 0 && this.LinkedEntityPrimaryKeys.length > 0) {
+      if (this.LinkedEntityID && this.LinkedEntityID > 0 && this.CompositeKeyIsPopulated()) {
         const dci = await md.GetEntityObject<DataContextItemEntity>('Data Context Items');
         dci.NewRecord();
         dci.DataContextID = dc.ID;
         if (this.LinkedEntity === 'User Views') {
           dci.Type = 'view';
-          dci.ViewID = this.LinkedEntityPrimaryKeys[0].Value;
+          dci.ViewID = this.LinkedEntityCompositeKey.KeyValuePairs[0].Value;
         }
         else if (this.LinkedEntity === 'Queries') {
           dci.Type='query';
-          dci.QueryID = this.LinkedEntityPrimaryKeys[0].Value;
+          dci.QueryID = this.LinkedEntityCompositeKey.KeyValuePairs[0].Value;
```

Comment:
@JS-BC  this is fine. But it makes me think we should have a public function on the CompositeKey class called `GetValue) where you can pass in a `Key` and receive null or value back.

Reply:
Good idea